### DISCUSSION TITLE
Can we create dedicated stream for a device

### DISCUSSION THREAD

**Discussion by kumardeepak5** [edgexfoundry/community#188] [General]
Hi Team,

I have installed  edgex foundry by using [docker-compose-no-scty.yml](https://github.com/edgexfoundry/edgex-compose/blob/v3.0/docker-compose-no-secty.yml).

Now i am following https://docs.edgexfoundry.org/3.0/examples/Ch-CommandingDeviceThroughRulesEngine/ to actuate device by creating rule on the rule engine.

further my queries are as below-
1. As guided into document a single stream is created to capture all devices data but can we create dedicated stream for each device?
2. In this example we are writing rule around resource which is of primitive type but what if we need to write rule for the resource which is JSON/Object type.
3. Please suggest me document where i can explore SQL sysntax which could be used to write more complex rules.

thanks
Deepak Kumar

**Comment by kumardeepak5** [comment]
> Hi Team,
> 
> I have installed edgex foundry by using [docker-compose-no-scty.yml](https://github.com/edgexfoundry/edgex-compose/blob/v3.0/docker-compose-no-secty.yml).
> 
> Now i am following https://docs.edgexfoundry.org/3.0/examples/Ch-CommandingDeviceThroughRulesEngine/ to actuate device by creating rule on the rule engine.
> 
> further my queries are as below-
> 
> 1. As guided into document a single stream is created to capture all devices data but can we create dedicated stream for each device?
> 2. In this example we are writing rule around resource which is of primitive type but what if we need to write rule for the resource which is JSON/Object type.
> 3. Please suggest me document where i can explore SQL sysntax which could be used to write more complex rules.
> 
> thanks Deepak Kumar

please comment on same

**Reply by kumardeepak5** [reply]
please comment on same

**Comment by kumardeepak5** [comment]
please reply on same

**Comment by cloudxxx8** [comment]
Your questions are all about eKuiper rules engine, not EdgeX.
Please check the eKuiper document and forum
https://ekuiper.org/docs/en/latest/
